lib/reply.js

```diff
@@ -193,11 +193,24 @@ function handleError (reply, error, cb) {
     statusCode = statusCode || 500
   } else if (error.status >= 400) {
     statusCode = error.status
+    if (error.status === 404) {
```

Comment:
Why duplicate this logic?
You can just check `statusCode` few lines below (before `if (statusCode < 400)`)

Reply:
It can't be done later, because https://github.com/fastify/fastify/blob/938f8a551b03735e5bafa81d4c6927d2f9615ed1/lib/reply.js#L191 set the `statusCode`  variable to 404 at the beginning, breaking a good chunk of our tests.